### ISSUE TITLE
Internet Explorer 11 Support with immer

### ISSUE THREAD

**Issue by computebender** [wbkd/react-flow#894]
Hello,
I am using React Flow to create interactive flow charts. React Flow has easy-peasy as a dependency, which in turn has immer as a dependency. I am currently struggling to find a way to enable ES5 support with the instance of immer easy-peasy uses. 

I have a sample project that demonstrates where I am currently stuck [https://github.com/computebender/flow-ie-support](https://github.com/computebender/flow-ie-support) 
As well as a [codesandbox](https://codesandbox.io/s/friendly-antonelli-bxhii) (although that doesn't really help with IE support)

The current issue I am facing is an error thrown only in IE (not Chrome or Firefox) because of immer: 
`SCRIPT5022: [Immer] Cannot use proxies if Proxy, Proxy.revocable or Reflect are not available`

From here I have found two potential paths to follow,

1. Try to polyfill for Proxy using Google's [proxy-polyfill](https://github.com/GoogleChrome/proxy-polyfill)
2. Enable es5 support in immer by running [enableES5](https://immerjs.github.io/immer/docs/installation)

**1 - Polyfill**
From what I have read, Proxy cannot be fully implemented using polyfills. From babeljs.io:

> Due to the limitations of ES5, Proxies cannot be transpiled or polyfilled. See support in [various JavaScript engines.](https://kangax.github.io/compat-table/es6/#test-Proxy)

Although I have seen posts suggesting to use the Chrome teams [proxy-polyfill](https://github.com/GoogleChrome/proxy-polyfill).
I have tried both just importing proxy-polyfill in the index.js entrypoint of my app,
```js
import 'proxy-polyfill';
```
and importing the proxy.min.js file directly,
```js
import 'proxy-polyfill/proxy.min.js';
```
Neither of these had an impact, and the error in IE remained.

**2 - enableES5()**
From what I've read, it should be very easy to enable es5 support for immer, which would remove the Proxy dependency. So to do this I added immer 7.0.9 (the same version specified in easy-peasy's package.js) as a direct dependency of my project. (Note `npm ls immer` was run on Ubuntu, as that's where I'm writing this, but testing is being done on Windows obviously)
```
flow-ie-support@0.1.0 /home/ben/Projects/flow-ie-support
├── immer@7.0.9 
├─┬ react-flow-renderer@8.8.0
│ └─┬ easy-peasy@4.0.1
│   └── immer@7.0.9  deduped
└─┬ react-scripts@4.0.2
  └─┬ react-dev-utils@11.0.2
    └── immer@7.0.9  deduped
```
I then added:
```js
import { enableES5 } from 'immer';
enableES5()
```
to my index.js entrypoint. This also had no impact on the same Proxy error showing up in internet explorer. I'm just wondering if anyone has any suggestions to proceed.

**Comment by moklick** [comment]
Hey @computebender 

it's strange that the `enableES5` function doesn't work. Unfortunately I don't have any other tips for you.

We are working on a next version without easy-peasy. I think it will be easier to get it to work with IE11. We will probably release it in the next one or two weeks.

**Comment by moklick** [comment]
We just released v9.0.0. That one comes without easy-peasy but just redux.

**Comment by moklick** [comment]
I will close this issue since we are not using easy-peasy and immer anymore.
